src/hashmap.c

```diff
@@ -1,51 +1,36 @@
-#define BUCKETS 1000000
-#define PRIME_SEED_1 17
-#define PRIME_SEED_2 97
-#define KEY_CHARS 128
-#define VALUE_CHARS 128
-#define NODE_SIZE sizeof(Node)
-
+#include "hashmap.h"
 #include <stdlib.h>
-#include <stdio.h>
 #include <stdbool.h>
 #include <string.h>
-#include <assert.h>
-
-/* A key-value node in a single-linked list. */
-typedef struct Node Node;
-
-struct Node {
-    Node* next;
-    char key[KEY_CHARS];
-    char value[VALUE_CHARS]; 
-};
 
+#define BUCKETS 1000000
+#define PRIME_SEED_1 17
+#define PRIME_SEED_2 97
 
+/* A key-value node in a single-linked list. */
 Node* create_sentinel_node(){
-    void* ptr = malloc(NODE_SIZE);
-    if (ptr == NULL){
+    Node* node = malloc(sizeof(Node));
+    if (!node){
         exit(EXIT_FAILURE);
-    } else {
-        Node* node = (Node*)ptr; // safely typecast the pointer
-        node->next = NULL;
-        strcpy(node->key, "\0");
-        strcpy(node->value, "\0");
-        return node;
     }
+    node->next = NULL;
+    strcpy(node->key, "\0");
+    strcpy(node->value, "\0");
+    return node;
 }
 
 
 size_t modular_hash(char* string) {
     size_t hash = 0;
     for (char* character=string; *character != '\0'; character++) {
-        hash += ((int)*character + PRIME_SEED_1) * PRIME_SEED_2; 
```

Comment:
`char` is an integer (sounds weird, I know) so you don't have to cast before using it for arithmetic

Reply:
yup!